### PR TITLE
Fix one IPAppliance test + fix possible recursion leak

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d2eae99b0feb1c80624650742c2ba64f5db5e3bd:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: utils/tests/test_ipappliance.py [hunk 1]
```diff
@@ -9,7 +9,7 @@ def test_ipappliance_from_address():
     address = '1.2.3.4'
     ip_a = IPAppliance(address)
     assert ip_a.address == address
-    assert ip_a.url == 'https://%s/' % address
```
Comment (seandst) [inline utils/tests/test_ipappliance.py]: I disagree. IPAppliance.url should instead make sure the URL is predictable, and always has a trailing slash.

File: utils/appliance.py [hunk 2]
```diff
@@ -444,8 +444,12 @@ def rest_api(self):
...
     def address(self):
         # If address wasn't set in __init__, use the hostname from base_url
-        parsed_url = urlparse(self.url or store.base_url)
-        return parsed_url.netloc
+        if getattr(self, "_url", None) is not None:
+            parsed_url = urlparse(self.url)
+            return parsed_url.netloc
+        else:
+            parsed_url = urlparse(store.base_url)
+            return parsed_url.netloc
```
Comment (seandst) [inline utils/appliance.py]:
```python
if getattr(self, "_url", None) is None:
  parsed_url = urlparse(store.base_url)
else:
  parsed_url = urlparse(self.url)
return parsed_url
```

That said, it seems like the problem here is `self.url` not always doing the right thing. Seems like that should be the place to look into store.base_url and set the address properly.

Comment (mfalesni) [inline utils/appliance.py]: there is a circular dependency and I am not entirely sure what to do with that.

**Comment by dajoRH** [comment]
Lint report for commit 905564c42d2daea6a37bd998685aa20fbd8c5bdf:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: utils/appliance.py (hunk 2 above)
Comment (seandst) [inline utils/appliance.py]: Can you easily describe the dependencies?

Comment (mfalesni) [inline utils/appliance.py]: pytest_store can look up the base_url from env, or if a current_appliance is set, then it asks current_appliance for url. The url looks for self.address which by default used to refer to store.base_url. I addded the or there but accessing hte url makes it go to infinite recursion. This fixes the issue by not triggering the url if we won't use it.

**Pull request merged**